src/coreclr/nativeaot/Runtime/thread.cpp

```diff
@@ -607,60 +609,95 @@ bool Thread::IsHijackTarget(void * address)
     return false;
 }
 
-bool Thread::Hijack()
+void Thread::Hijack()
```

Comment:
Can this be called on a thread that is hijacked already - how is this case handled? (I am not able to tell by just looking at the code.)

Reply:
Trying to hijack a thread that is already hijacked is a common scenario. 

- if we can just suspend (inline or by redirection), we will do that. 
We will also remove the hijack in such case. We do not have to, but we know the hijack is no longer useful.
- if we get a new hijack address that is different from existing one, we assume the new one is better (it must be deeper in the call-graph), so we remove the old hijack and add a new one.
- if the new hijack is the same as old, we do nothing.

One thing to be observed is that having a hijack on a thread is not harmful - 
If a thread returns and hits a hijack, it will either suspend or self-unhijack.
If some thread needs to do a stackwalk, it will remove the hijack.
If we want to place a hijack, we can remove the old one.